src/main/java/faang/school/godbless/distributedArmyOfHeroes/Army.java

```diff
@@ -0,0 +1,36 @@
+package faang.school.godbless.distributedArmyOfHeroes;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Army {
+    private List<Subdivision> subdivisions;
+
+    public Army() {
+        this.subdivisions = new ArrayList<>();
+    }
+
+    public void addUnit(Subdivision subdivision) {
+        subdivisions.add(subdivision);
+    }
+
+    public int calculateTotalPower() throws InterruptedException {
+        List<SubdivisionThred> subdivisionThreds = new ArrayList<>();
+        List<Integer> power = new ArrayList<>();
+
+        for (Subdivision subdivision : subdivisions) {
+            SubdivisionThred subdivisionThred = new SubdivisionThred(subdivision);
+            power.add(subdivision.getPower());
+            subdivisionThreds.add(subdivisionThred);
+            subdivisionThred.run();
+        }
+
+        for (SubdivisionThred subdivisionThred : subdivisionThreds) {
+            subdivisionThred.getSubdivision().join(); //тут вопрос
```

Comment:
List<Thread> threads = new ArrayList<>();
        List<SubdivisionThred> subdivisionThredList = new ArrayList<>();

        for (Subdivision subdivision : subdivisions) {
            SubdivisionThred subdivisionThred = new SubdivisionThred(subdivision);
            Thread thread = new Thread(subdivisionThred);
            threads.add(thread);
            subdivisionThredList.add(subdivisionThred);
            thread.start();
        }

        for (Thread thread : threads) {
            thread.join();
        }

        return subdivisionThredList.stream()
                .mapToInt(SubdivisionThred::getPowerSubdivision)
                .sum();